tests/python/test_dataset_aggregation.py

```diff
@@ -0,0 +1,139 @@
+# BSD 2-Clause License
```

Comment:
Because there is a copy operation (or potentially move if we update the routines) outside of the core C++ layer, should we test in this test that the retrieved datasets are in fact the same as the sent ones?  It might be a good sanity check.